pkg/account/accounts.go

```diff
@@ -0,0 +1,34 @@
+package account
+
+import (
+	"github.com/Optum/dce/pkg/errors"
+	"github.com/Optum/dce/pkg/model"
+	validation "github.com/go-ozzo/ozzo-validation"
+)
+
+// Accounts is a list of type Account
+type Accounts struct {
+	data model.Accounts
+}
+
+// GetAccounts Get a list of accounts based on Principal ID
+func GetAccounts(q *model.Account, d MultipleReader) (*Accounts, error) {
+	err := validation.ValidateStruct(q,
+		// ID has to be empty
+		validation.Field(&q.ID, validation.NilOrNotEmpty, validation.By(isNil)),
+	)
+	if err != nil {
+		return nil, errors.NewValidation("account", err)
```

Comment:
What will this validation error look like to the end user?
Thinking that an API response like `"account validation error: ID must not be nil"` would be pretty confusing.
Maybe better to do our own check (here, or in controller), and response like `"'id' is not a supported query parameter"`

Reply:
Updated.